test/hdt-test.js

```diff
@@ -873,6 +943,23 @@ describe('hdt', function () {
       });
     });
 
+    describe('reading the Header', function () {
+      var triples;
+      before(function () {
+        return document.readHeader().then(result => {
+          triples = result;
+        });
+      });
+      it('should return an array with matches', function () {
+        triples.should.be.an.Array();
+        triples.should.have.length(28);
+        triples[0].should.eql({
+          subject: '<file://literals.ttl>',
```

Comment:
We can't do this format, it's inconsistent with the rest of the library. If we return triples, they should be [N3.js in-memory representation](https://github.com/RubenVerborgh/N3.js#triple-representation) like the `searchTriples` method.

Alternatively, we don't return an array of triples, but a text stream or string (which can then be parsed through a library of choice.)

Reply:
Good point. Reading this as a string should be quite easy (almost just a matter of concatenating the terms). writing as a string is a bit more tricky though as we'd have to convert from a string to this particular triply format. Not sure what's the best solution here